### PR TITLE
Best branch

### PR DESCRIPTION
**Pull request by omerza**
Conv backprop is vectorized.
Added - 
BatchNorm
Momentum

### CODE REVIEW

**Comment by omerza** [comment]
Added conv backprop numpy, momentum, NN base class, Leaky Relu, Tanh
